operator/v1/stable.ingresscontroller.testsuite.yaml

```diff
@@ -13,3 +13,390 @@ tests:
+  - name: Should not allow to set/delete dynamic request headers values with not allowed converters.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:          
+            request:
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,hello]"
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should not allow to set/delete dynamic request headers values with not allowed sample fetchers.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-not-allowed-values
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:          
+            request:
+            - name: X-SSL-Client-Cert 
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der1234,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+  - name: Should be able to add set field only when discriminant type is Set.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+    expectedError: "set is required when type is Set, and forbidden otherwise"
```

Comment:
done

Reply:
"set is required when type is Set, and forbidden otherwise"     ------can't be done as type is a required field and will result in error:

Message: "IngressController.operator.openshift.io \"test-279v6\" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]",

Reply:
Can you add the test with `expectedError: 'Invalid value: "null"'`?

Reply:
ok done

Reply:
The CEL validation is wrong if you are getting the correcting existing errors validation. Updating the test is not the correct move. You have to check for the presence of the field before you assess its value

Reply:
The problem might be that `has(self.foo)` returns true even if `foo` is null ([playground](https://playcel.undistro.io/?content=H4sIAAAAAAAAA6tWSkksSVSyUipOzUmzislTUCipLEi1UghOLQFxilNLrBTySnNylHSUUisKilKLizPz84DKMxKLNUBa9EDKNRXU1BTgPAVbWwV1oH51BXsFBbg6oEmaClYKiigCSrUACumMKoAAAAA%3D)).

`(has(self.type) && self.type == 'Set') == (has(self.set) && self.set != null)` seems to do the right thing ([playground](https://playcel.undistro.io/?content=H4sIAAAAAAAAA6tWSkksSVSyUipOzUmzislTUCipLEi1UghOLQFxilNLrBTySnNylHSUUisKilKLizPz84DKNTISizVAevRA6jUV1NQU4DwFW1sFdaAB6pogFkIl0DCEQiBHQdEWbLamUi0AwpQpy4cAAAA%3D)).

Reply:
`set` has omitempty here so it should not be present when it's `null`, but I take you're point 🤔 Looking at the test cases, those are also unstructured and not including `set` so I'm not sure that is the issue

Reply:
So I had a quick look at this today and I think the CEL is ok as it. The issue here is that there are multiple errors, 
```
"Route.route.openshift.io \"hello-openshift-actions\" is invalid: [spec.httpHeaders.actions.response[0].action.type: Required value, <nil>: Invalid value: \"null\": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation]",
```
The first is that you've omitted a required field, the second tells you to fix the other errors first, if you fix the first error, then in this case, the second error goes away. 

The tests should never have the `some validation rules were not checked because the object was invalid` as the expected error, but, if there's another error in there, then I think it's ok to have this

Reply:
Discussed on Slack; the test is fine as-is.